control-plane/connect-inject/envoy_sidecar.go

```diff
@@ -27,25 +28,27 @@ func (w *MeshWebhook) envoySidecar(namespace corev1.Namespace, pod corev1.Pod, m
 		containerName = fmt.Sprintf("%s-%s", envoySidecarContainer, mpi.serviceName)
 	}
 
-	container := corev1.Container{
-		Name:  containerName,
-		Image: w.ImageEnvoy,
-		Env: []corev1.EnvVar{
-			{
-				Name: "HOST_IP",
-				ValueFrom: &corev1.EnvVarSource{
-					FieldRef: &corev1.ObjectFieldSelector{FieldPath: "status.hostIP"},
-				},
+	probe := &corev1.Probe{
+		Handler: corev1.Handler{
+			TCPSocket: &corev1.TCPSocketAction{
+				Port: intstr.FromInt(EnvoyInboundListenerPort + mpi.serviceIndex),
 			},
 		},
+		InitialDelaySeconds: 1,
+	}
```

Comment:
This probe is to replace our existing TCP health check that we register with consul. I didn't add startup probe as we don't have anything like that for now and so I'm not sure if adding a startup probe will cause problems because we haven't see envoy being slow to start up.